onmt/modules/Sequencer.lua

```diff
@@ -26,6 +28,16 @@ end
 function Sequencer:_sharedClone()
   local clone = self.network:clone('weight', 'gradWeight', 'bias', 'gradBias')
 
+  local cloneRNN = onmt.utils.Tensor.findModule(clone, self.rnnClass)
+  if cloneRNN ~= nil then
+    cloneRNN:shareRecurrents(onmt.utils.Tensor.findModule(self.network, self.rnnClass))
+  end
+  function clone:setTimeStep(t)
```

Comment:
Ok, I thought that was a little hacky but if its acceptable it's definitely a better solution. 

I overlooked the second fact - that might be what's leading to your NaN perplexity on validation. I will fix this in a second and then push to this PR.